tests/test_dojo.py

```diff
@@ -147,4 +149,69 @@ def test_print_unallocated_filename(self):
         self.dojo.add_person('bett', 'fellow')
         self.dojo.add_person('khalid', 'fellow', 'Y')
         self.dojo.print_unallocated('unallocated') 
-        self.assertFalse(os.path.isfile('unallocated.txt'))      
\ No newline at end of file
+        self.assertFalse(os.path.isfile('unallocated.txt'))
+        self.assertTrue(os.path.getsize('alocated.txt') > 0)
+        self.assertTrue(os.path.exists('alocated.txt'))
+
+    #    tests for reallocate person starts here --->
+    def test_reallocate_person(self):
+        '''test that a person has been reallocated to a different room'''
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.add_person('john', 'staff')
+        staff = self.dojo.staff[0]
+        office_name = staff.office.name
+        self.assertEqual(office_name, 'hogwarts')
+        self.dojo.create_room('swift', 'office')
+        self.dojo.reallocate_person('joshua', 'swift')
+        new_office_name = staff.office.name
+        self.assertEqual(new_office_name, 'swift')
+
+    def test_if_reallocated_to_the_same_room(self):
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.add_person('john', 'staff')
+        staff = self.dojo.staff[0]
+        office_name = staff.office.name
+        self.assertEqual(office_name, 'hogwarts')
+        self.dojo.reallocate_person('joshua', 'hogwarts')
+        new_office_name = staff.office.name
+        self.assertEqual(new_office_name, 'cant reallocate person to the same room')
```

Comment:
new_office_name holds the name of office, you comparing the office name to the message stated, they cant be equal, i think you should assert with the message of line 175